### PR TITLE
Bring us into compliance with YouTube API terms

### PR DESCRIPTION
**Pull request by mia-pi-git**
Three main issues have been resolved here.
1 - Now, only one YouTube player can ever be on screen at once. Old ones are removed when the new one is added. [relevant section - III.C 1](https://developers.google.com/youtube/terms/developer-policies#c.-implementing-youtube-features)
2 - We didn't properly disclose we were using YouTube API services [relevant sections - III.A 2 B](https://developers.google.com/youtube/terms/developer-policies#a.-api-client-terms-of-use-and-privacy-policies)
3- We didn't disclose it allowed third-party content/advertisements. [relevant sections - III.A 2 F](https://developers.google.com/youtube/terms/developer-policies#a.-api-client-terms-of-use-and-privacy-policies)

(re: index.template.html change, terms state the privacy policy must be prominently displayed and accessible. An easily-missed command does not fulfill this either.)
This needs to be merged within 7 days of creation, and we must restart to deploy it.

### CODE REVIEW

**Review by HoeenCoder — APPROVED** [review]

File: src/battle-log.ts [hunk 1]
```diff
@@ -1016,6 +1018,66 @@ export class BattleLog {
...
+					onStateChange: (event: any) => {
+						if (event.data === window.YT.PlayerState.PLAYING) {
+							for (const curPlayer of BattleLog.players) {
+								if (player.psId === curPlayer.psId) continue;
+								curPlayer.pauseVideo?.();
+							}
+						}
+					},
+					onReady: () => {
+						// scroll chat - if it's a page it shouldn't matter much regardless
```
Comment (Zarel) [inline src/battle-log.ts:1045]: PS intentionally only scrolls if it's already at the bottom which this utterly disregards :[

File: src/battle-log.ts [hunk 2]
```diff
@@ -1016,6 +1018,66 @@ export class BattleLog {
 		this.localizeTime);
 	}
 
+	static initYoutubePlayer(
+		id: string, dim: [string, string], videoId: string, time: number | null = null
+	) {
+		const [height, width] = dim;
+		const loadPlayer = () => {
+			const el = $(`div[data-playerid=${id}]`).get(0);
```
[the remark this answers is not in the document]
Comment (mia-pi-git) [reply src/battle-log.ts:1026]: This function is in timeouts to ensure it's run after the element gets placed. I'll swap it to a class though.

File: src/battle-log.ts (hunk 1 above)
Comment (mia-pi-git) [reply src/battle-log.ts:1045]: OK. Will remove.

File: src/battle-log.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Zarel) [reply src/battle-log.ts]: It's not, though. I checked before writing this post, it's not in timeouts.

**Comment by Zarel** [comment]
This is vulnerable to race conditions. Not just hypothetical, either, it will definitely fail if two YouTube videos are in scrollback when you open PS.

**Comment by Zarel** [comment]
🚀🚀🚀

**Pull request merged**